openmc/data/resonance_covariance.py

```diff
@@ -0,0 +1,743 @@
+            used to import values not contained in File 32
+
+        Returns
+        -------
+        openmc.data.ResonanceCovariances
+            Resonance covariance data
+
+        """
+        file_obj = io.StringIO(ev.section[32, 151])
+
+        # Determine whether discrete or continuous representation
+        items = endf.get_head_record(file_obj)
+        n_isotope = items[4]  # Number of isotopes
+
+        ranges = []
+        for iso in range(n_isotope):
+            items = endf.get_cont_record(file_obj)
+            abundance = items[1]
+            fission_widths = (items[3] == 1)  # Flag for fission widths
+            n_ranges = items[4]  # Number of resonance energy ranges
+
+            for j in range(n_ranges):
+                items = endf.get_cont_record(file_obj)
+                # Unresolved flags - 0: only scattering radius given
+                #                    1: resolved parameters given
+                #                    2: unresolved parameters given
+                unresolved_flag = items[2]
+                formalism = items[3]  # resonance formalism
+
+                # Throw error for unsupported formalisms
+                if formalism in [0, 7]:
+                    error = 'LRF = '+str(formalism)+'covariance not supported '\
+                            'for this formalism'
+                    raise NotImplementedError(error)
+
+                if unresolved_flag in (0, 1):
+                    # Resolved resonance region
+                    resonance = resonances.ranges[j]
+                    erange = _FORMALISMS[formalism].from_endf(ev, file_obj,
+                                                              items, resonance)
+                    ranges.append(erange)
+
+                elif unresolved_flag == 2:
+                    warn = 'Unresolved resonance not supported. Covariance '\
+                           'values for the unresolved region not imported.'
+                    warnings.warn(warn)
+
+        return cls(ranges)
+
+
+class ResonanceCovarianceRange:
+    """Resonace covariance range. Base class for different formalisms.
+
+    Parameters
+    ----------
+    energy_min : float
+        Minimum energy of the resolved resonance range in eV
+    energy_max : float
+        Maximum energy of the resolved resonance range in eV
+
+    Attributes
+    ----------
+    energy_min : float
+        Minimum energy of the resolved resonance range in eV
+    energy_max : float
+        Maximum energy of the resolved resonance range in eV
+    parameters : pandas.DataFrame
+        Resonance parameters
+    covariance : numpy.array
+        The covariance matrix contained within the ENDF evaluation
+    lcomp : int
+        Flag indicating format of the covariance matrix within the ENDF file
+    mpar : int
+        Number of parameters in covariance matrix for each individual resonance
+    formalism : str
+        String descriptor of formalism
```

Comment:
Need to add description of file2res here (since it is assumed to be present for any subclass)